addons/sale/sale.py

```diff
@@ -117,6 +120,7 @@ def _compute_tax_id(self):
     pricelist_id = fields.Many2one('product.pricelist', string='Pricelist', required=True, readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, help="Pricelist for current sales order.")
     currency_id = fields.Many2one("res.currency", related='pricelist_id.currency_id', string="Currency", readonly=True, required=True)
     project_id = fields.Many2one('account.analytic.account', 'Analytic Account', readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, help="The analytic account related to a sales order.", copy=False)
+    related_project_id = fields.Many2one('account.analytic.account', inverse='_inverse_project_id', related='project_id', string='Analytic Account', help="The analytic account related to a sales order.", copy=False)
```

Comment:
Yes yes I understood why you did it but it seemed a bit like a hack. After talking with @rco-odoo , it appears that there's is no other solution and that what you did is not really a hack (having a related field to itself); so your implementation is ok